File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        if(root == null) return 0;
        return dfs(root, 0) - 1;
    }

    public int dfs(TreeNode root, int level){
        if(root == null){
            return 1;
        }

        int left = dfs(root.left, level) + 1;
        int right = dfs(root.right, level) + 1;
        
        return Math.max(right, left);

    }
}

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        
        res = []
        self.dfs(root, res)
        return res

    def dfs(self, root: Optional[TreeNode], res: List[int]) -> None:
        if root is None:
            return
        
        if root.left:
            self.dfs(root.left, res)
        res.append(root.val)
        if root.right:
            self.dfs(root.right, res)